primitives/src/authorization.rs

```diff
@@ -86,6 +88,16 @@ pub struct Authorization<U> {
     pub auth_id: u64,
 }
 
+/// Authorization Identity data
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug, PartialOrd, Ord)]
+pub struct JoinIdentityData {
```

Comment:
We can implement From<Identity> for this to avoid passing empty vecs of permissions at various places.